### ISSUE TITLE
Service mode for Assistant

### ISSUE THREAD

**Issue by ramiresviana** [Ylianst/MeshCentralAssistant#13]
The Assistant is very useful for remote support, end users do not require elevated privileges to run the application, but there'll be some cases that running an program with admin rights or switching between users will trigger UAC, blocking the interaction with the remote screen. That means that Assistant require some privileges to bypass the UAC, but the problem still happens when running it with admin rights. This problem can be circumvented by having MeshAgent installed and running as service, this means having the agent available all the time, even not being needed, that is the pros of using Assistant for remote support.

My request is the addition of an service mode for the Assistant, allowing it to bypass the UAC, and uninstalling the service automatically after closing the application. Since installing a service requires elevated privileges, the Assistant must run as service only when requested.

The process of accessing an remote machine and running Assistant on service mode would be like:

1. End user, without elevated privileges, downloads the Assistant and runs it on user mode.
2. The operator connects to the user machine and requests the Assistant to run on service mode.
3. The Assistant requests admin credentials and uses runas to run it on service mode.
4. The operator can now run programs without UAC blocking the desktop.
5. The operator ends the session by closing the Assistant and the service is uninstalled from the machine.

This could also fix https://github.com/Ylianst/MeshCentral/issues/1885

**Comment by Ylianst** [comment]
Agreed that being able to capture UAC prompt would be great. Getting Assistant to run as a service is a big change, but I may be able to capture the UAC prompt when Assistant is run as administrator and that will be the next step.

**Comment by tuzumkuru** [comment]
Can this workflow achieved now? I wanted to ask since it has been some time. Otherwise +1 for this feature request.

I am in need of a similar workflow and was thinking of exactly this solution before I saw this issue.

**Comment by si458** [comment]
@tuzumkuru you can run the normal agent and click connect then accept the uac prompt and runs in like a portable mode where when u quit the support ends
This might be more ideal for u
Or just install the meshagent as a windows server on the background for 24.7 access
The assistant is designed for control without uac prompts and being run as the user and NOT as an admin

**Comment by tuzumkuru** [comment]
@si458 Thanks for your input—I really appreciate it, and thanks to everyone supporting the project. 

I understand your perspective, but some workflows aren’t conveniently supported:

- I don’t want an always-on, privileged client on employees’ computers.
- Ideally, I’d like to connect with full privileges after their request.
- I also want to be able to see devices with their basic info without having admin access.

The UAC prompt can’t be handled remotely without someone on the employee’s device accepting it.

**MeshAgent in Connect mode:** limited access and no easy way to see online devices with basic info.
**MeshAgent in Service mode:** always-on with full privileges, which may feel insecure for employees (the remote users).
**MeshCentral Assistant:** Elevated access and UAC prompt issue remains.

In short, while each tool has useful features, this workflow—common in other remote desktop software like TeamViewer or AnyDesk—isn’t achievable with the current setup.

**Comment by si458** [comment]
@tuzumkuru why not just enable the userconsent then for the device group or even for the full domain/server?

Then when a staff member needs to connect to a remote device (clients pc) the client has to click accept first?

This can all be done with the meshagent installed as a service

![Image](https://github.com/user-attachments/assets/5337323d-cdb2-474d-aacc-e22bf80b1f65)

**Comment by tuzumkuru** [comment]
@si458 It still doesn’t prevent the system admin from connecting whenever they want, since they control the system. (I’m the system admin here, but I’m trying to look at it from the user’s perspective.) Knowing that an admin could silently watch your computer without any trace doesn’t feel very secure.

Another use case: imagine we’re producing machines and want to access them with full privileges only when help is needed. From the user’s perspective, would you feel comfortable with software that’s always running on your machine and connected to your network?

That’s where MeshCentral Assistant feels closer to the right workflow:

You can stay connected and the server still has your basic info.

You’re safe, since nobody can connect unless you allow it.

If someone requests access, it asks for your permission.

“Call for help” is great (and would be even better if it prevented the computer from going to sleep), enabling connections on demand.

The only real drawback is that it doesn’t have system-level privileges. This could be solved if “call for help” also spun up a temporary MeshAgent service with full privileges, which could then be disposed of after use.

**Comment by si458** [comment]
@tuzumkuru if u enable the userconsentflag option in config.json, it applies/forces consent TO ALL USERS INCLUDING THE ADMIN?

**Comment by tuzumkuru** [comment]
@si458 I understand, but what about the system admin that can also change the config.json? From the user perspective it is still problematic. 

And is it possible to force consent to all users but not some groups (ie, servers that there is nobody as a user)?

**Comment by si458** [comment]
Are u saying ur worried that u as the server admin and administrator of your meshcentral server could change the consent in the config.json? 
I would be more worried a hacker got into my meshcentral server and changed the config.json than myself just changing it as a one off?

Also the way to mix consent and no consent is to not set consent, then create a user group, add ur staff into it, then set the user group to enforce consent for all their access

Then when u connect u have choice ask for consent (right click connent and click ask for consent) or just click connect and don't have consent

The is also other options that have been added recently too which if u enforce consent but say the is no user logged in, then u can connect and do what u need because the is no user actually using the desktop!

**Comment by si458** [comment]
@tuzumkuru if u want tomorrow (12hours time im uk based utc+1 at mo) give me a fleemail/disc/tele/whats/tweet/sig and we can discuss ur concerns u have and maybe work something out for u as meshcentral has LOADS of features, it's just getting everything setup how u want it etc :)

**Comment by tuzumkuru** [comment]
@si458 sent you an email for further discussion :D
